lib/core/models/repositories/roleRepository.js

```diff
@@ -181,25 +194,130 @@ class RoleRepository extends Repository {
    * @returns Promise
    */
   validateAndSaveRole (role, options) {
-    return role.validateDefinition()
+
+    return role
+      .validateDefinition()
       .then(() => {
         if (role._id === 'anonymous' && !role.canLogIn()) {
           errorsManager.throw('security', 'role', 'login_required');
         }
 
-        this.kuzzle.emit(
-          'core:roleRepository:save',
-          { _id: role._id, controllers: role.controllers });
-
-        return this.persistToDatabase(role, options);
+        this.kuzzle.emit('core:roleRepository:save', {
+          _id: role._id,
+          controllers: role.controllers
+        });
+      })
+      .then(() => this.checkRoleNativeRights(role))
+      .then(() => {
+        const force = options ? options.force : false;
+        this.checkRolePluginsRights(role, force);
       })
+      .then(() => this.persistToDatabase(role, options))
       .then(() => this.loadOneFromDatabase(role._id))
       .then(updatedRole => {
         this.roles.set(role._id, updatedRole);
         return updatedRole;
       });
   }
 
+  /**
+   * Given a Role object, checks if its controllers and actions exist.
+   *
+   * @param {Role} role
+   */
+  checkRoleNativeRights(role) {
+    const roleRightsError = errorsManager.wrap('security', 'role');
+    Object.keys(role.controllers).forEach(roleController => {
+      if (roleController !== '*' && !this.kuzzle.funnel.controllers.has(roleController)) {
+        if (roleController.split('/').length > 1) {
+          // Meaning this is a plugin controller. So we skip it.
+          return;
+        }
+        roleRightsError.throw('unknown_controller', role._id, roleController);
+      }
+      if (roleController === '*') {
+        this.kuzzle.funnel.controllers.forEach(controller => {
```

Comment:
I don't see the point to check if an action really exists on all controllers mostly because I don't see the usecase here. I think it's better to forbid this kind of rights.

```js
controllers: {
  '*': {
    actions: {
      create: true // all controllers must have a "create" action otherwise this role is invalid
    }
  }
}
```

We should only accept the a wildcard controller with a wildcard action